### DISCUSSION TITLE
Tresca yielding criteria

### DISCUSSION THREAD

**Discussion by rh201** [idaholab/moose#24864] [Q&A Modules: Solid mechanics]
Dear MOOSEers,

Are there any tresca-yielding-criteria related documents/codes implemented in MOOSE? The plasticity modulus in MOOSE now is using Von-Mises yielding criteria, I want to use Tresca one, how should it be adapted to it? Thanks.

**Comment by hugary1995** [comment]
This may be a question for @jasondhales, @bwspenc and @jiangwen84. Before they reply you with a more definitive answer, I think it will be difficult because many classes in the tensor_mechanics are hard-coded with the Mises flow. I often just write a plasticity model from scratch if I need tresca's flow (the radial return utilities are still applicable in that case).

**Reply by rh201** [reply]
Thanks @hugary1995 , could you kindly share the code of it?  I tried to adapt the von-mises one to Tresca, where I changed the yielding function,  but it seems not work.

**Reply by rh201** [reply]
@hugary1995 Or which part should be changed from von-mises modules? Thanks.
